### PR TITLE
Split trace agreggation

### PR DESCRIPTION
**Pull request by mapno**
#### PR Description

Split trace aggregation from tail sampling, as it can be useful for other purposes too, such as service graphs or other processors that benefit from having complete traces.

The pipeline will use `groupbytrace` processor for aggregation, except when using tail sampling, in which case grouping will be performed by the embedded aggregator in the sampling processor. It's not possible to have the tail sampling processor only sampling and leave aggregation to `groupbytrace`. Tail sampling has to buffer traces for at least 1s, so it's just better not to use `groupbytrace` in that case for simplicity and performance.

#### PR Checklist

- [x] CHANGELOG updated 
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by mapno** [comment]
There are a few things still missing from the PR, mostly documentation, so I marked it as a draft.

**Comment by mapno** [comment]
It's ready for review now

File: docs/configuration/tempo-config.md [hunk 1]
```diff
@@ -226,6 +221,24 @@ tail_sampling:
...
+# This is of particular interest for processor that benefit from having complete
+# traces, such as tail-based sampling
+#
+# A trace will be considered complete after the defined `wait` duration has
+# passed since its first span arrived to the processors.
+# If the trace is not complete by then, it'll be split into more than one trace.
+#
+# Longer waiting times will increase the number of traces that are correctly
+# grouped. However, it will also increase the memory overhead of the processor.
+group_by_trace:
```
Comment (joe-elliott) [inline docs/configuration/tempo-config.md:235]: shouldn't the "load balancing" settings be moved out as well? maybe renamed something to suggest it's routing by trace id.

File: docs/configuration/tempo-config.md [hunk 2]
```diff
@@ -226,6 +221,24 @@ tail_sampling:
...
+  
+  # Defines the time to wait for a complete trace before considering it complete
+  [ wait: <duration> | default="5s" ]
+    
+  # Configures the max amount of traces to keep in memory waiting for the duration
+  [ num_traces: <int> | default="1_000_000" ]
```
Comment (joe-elliott) [inline docs/configuration/tempo-config.md:241]: max_traces?

**Pull request closed without merging**